src/core/Rat.pm6

```diff
@@ -35,6 +71,28 @@ my class FatRat is Cool does Rational[Int, Int] {
     multi method perl(FatRat:D:) {
         "FatRat.new($!numerator, $!denominator)";
     }
+    multi method Str(FatRat:D:) {
+        my $whole  = self.abs.floor;
+        my $fract  = self.abs - $whole;
+
+        my $result = nqp::if(
+            nqp::islt_I($!numerator, 0), '-', ''
+        ) ~ $whole;
+
+        if $fract {
+            if $!denominator.chars < 5 {
```

Comment:
Can't this be just `$!denominator < 10000`? That would be faster than converting it to a string to count the chars.

Reply:
Good point. I'll fix that.